src/components/MapView/Legends/index.tsx

```diff
@@ -154,6 +174,104 @@ function LegendItem({
     }
   };
 
+  const boundaryLayer = getBoundaryLayerSingleton();
+  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
+    | LayerData<BoundaryLayerProps>
+    | undefined;
+
+  const legendLayerData = useSelector(layerDataSelector(layer.id)) as
+    | LayerData<LayerType>
+    | undefined;
+
+  const { startDate: selectedDate } = useSelector(dateRangeSelector);
+
+  const downloadToFile = (
+    source: { content: string; isUrl: boolean },
+    fileName: string,
+    contentType: string,
+  ) => {
+    const link = document.createElement('a');
+    link.setAttribute(
+      'href',
+      source.isUrl
+        ? source.content
+        : URL.createObjectURL(
+            new Blob([source.content], { type: contentType }),
+          ),
+    );
+
+    link.setAttribute('download', fileName);
+    link.click();
+  };
+
+  const handleLayerDownload = (
+    legendLayer: LayerType,
+    e: React.ChangeEvent<{}>,
+  ) => {
+    e.preventDefault();
+    const extent = bbox(boundaryLayerData?.data) as Extent;
+
+    switch (legendLayer.type) {
+      case 'wms': {
+        const dataUrl = getWCSLayerUrl({
+          layer: legendLayer,
+          extent,
+          date: selectedDate,
+        });
+        downloadToFile(
```

Comment:
does download related code belong in `Legends`? Moving this to a new util file would be nice for organization and if we want to support downloading through other sources.

Reply:
This makes perfect sense. Since it can be used somewhere else

Reply:
I agree, let's move these util functions somewhere else.